cloud/deployment/inspect/inspect.go

```diff
@@ -0,0 +1,157 @@
+package inspect
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/astronomer/astro-cli/astro-client"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/astronomer/astro-cli/config"
+)
+
+type deploymentInfo struct {
+	DeploymentID   string    `mapstructure:"deployment-id"`
+	WorkspaceID    string    `mapstructure:"workspace-id"`
+	ClusterID      string    `mapstructure:"cluster-id"`
+	AirflowVersion string    `mapstructure:"airflow-version"`
+	ReleaseName    string    `mapstructure:"release-name"`
+	DeploymentURL  string    `mapstructure:"deployment-url"`
+	WebserverURL   string    `mapstructure:"webserver-url"`
+	UpdatedAt      time.Time `mapstructure:"updated-at"`
+	Status         string    `mapstructure:"status"`
+}
+
+type deploymentConfig struct {
+	Name              string              `mapstructure:"name"`
+	Description       string              `mapstructure:"description"`
+	ClusterID         string              `mapstructure:"cluster-id"` // this is also in deploymentInfo
+	RunTimeVersion    string              `mapstructure:"runtime-version"`
+	SchedulerAU       int                 `mapstructure:"scheduler-au"`
+	SchedulerReplicas int                 `mapstructure:"scheduler-replicas"`
+	AlertEmails       []string            `mapstructure:"alert-emails"`
+	WorkerQueues      []astro.WorkerQueue `mapstructure:"worker-queues"`
+	AstroVariables    Variables           `mapstructure:"astronomer-variables"`
+}
+
+type PrintableDeployment struct {
+	DeploymentInfo   map[string]interface{} `mapstructure:"deployment-info"`
+	DeploymentConfig map[string]interface{} `mapstructure:"deployment-config"`
+}
```

Comment:
nitpick: couldn't find any reference to these structs, are they relevant? although I generally prefer getting data into the structs and then marshaling them for output

Reply:
They are relevant because that is what we want to output regardless of YAML or JSON format. Their keys are different than a deployment for e.g. `deployment.ClusterID` becomes `deployment.cluster-id` when we print the output. Currently the tests marshal them into structs to validate that we created them as expected. I did try marshaling them in `formatPrintableDeployment()` before printing them but then keys were not as expected. 

Eventually as part of this epic, we are going to allow users to create a deployment by passing in a `deploymentConfig` in YAML format at which point we will marshal that input into a `deploymentConfig` struct before using it. We switched the inspect vs. CRUD order a little bit to get early feedback from users.